fractions/javaee/jaxrs/src/main/java/org/wildfly/swarm/jaxrs/runtime/DefaultApplicationDeploymentProcessor.java

```diff
@@ -0,0 +1,173 @@
+package org.wildfly.swarm.jaxrs.runtime;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.ws.rs.core.Application;
+
+import org.jboss.shrinkwrap.api.Archive;
+import org.jboss.shrinkwrap.api.ArchiveEvent;
+import org.jboss.shrinkwrap.api.ArchiveEventHandler;
+import org.jboss.shrinkwrap.api.ArchivePath;
+import org.jboss.shrinkwrap.api.ArchivePaths;
+import org.jboss.shrinkwrap.api.Node;
+import org.jboss.shrinkwrap.api.asset.ArchiveAsset;
+import org.jboss.shrinkwrap.api.asset.Asset;
+import org.jboss.shrinkwrap.api.asset.ByteArrayAsset;
+import org.objectweb.asm.ClassReader;
+import org.wildfly.swarm.config.runtime.AttributeDocumentation;
+import org.wildfly.swarm.jaxrs.JAXRSArchive;
+import org.wildfly.swarm.jaxrs.JAXRSMessages;
+import org.wildfly.swarm.jaxrs.internal.ApplicationFactory2;
+import org.wildfly.swarm.jaxrs.internal.ApplicationPathAnnotationSeekingClassVisitor;
+import org.wildfly.swarm.spi.api.Defaultable;
+import org.wildfly.swarm.spi.api.DeploymentProcessor;
+import org.wildfly.swarm.spi.api.annotations.Configurable;
+import org.wildfly.swarm.spi.runtime.annotations.DeploymentScoped;
+import org.wildfly.swarm.undertow.descriptors.WebXmlAsset;
+
+@DeploymentScoped
+public class DefaultApplicationDeploymentProcessor implements DeploymentProcessor {
+
+    /**
+     * Path to the web.xml descriptor.
+     */
+    private static final ArchivePath PATH_WEB_XML = ArchivePaths.create(ArchivePaths.create("WEB-INF"), "web.xml");
+
+    @AttributeDocumentation("Set the JAX-RS application path, if the JAX-RS application class is autogenerated by Swarm")
+    @Configurable("swarm.deployment.*.jaxrs.application-path")
```

Comment:
This is the web context?

I think it might be worthwhile to use `swarm.deployment.*.context.path` same as 

https://github.com/wildfly-swarm/wildfly-swarm/blob/master/fractions/javaee/undertow/src/main/java/org/wildfly/swarm/undertow/runtime/ContextPathArchivePreparer.java#L40

If this could be a sub-path of the context path, then ignore me.